boto3/iam/inject.py

```diff
@@ -0,0 +1,27 @@
+# Copyright 2015 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You
+# may not use this file except in compliance with the License. A copy of
+# the License is located at
+#
+# http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is
+# distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific
+# language governing permissions and limitations under the License.
+
+from boto3 import utils
+
+def inject_access_key_methods(class_attributes, **kwargs):
+    utils.inject_attribute(class_attributes, 'load', access_key_load)
+
+def access_key_load(self, *args, **kwargs):
+    """
+    Call s3.client.list_access_keys to get the status
+    """
+    response = self.meta.client.list_access_keys(UserName= self.user_name)
+    for access_key in response['AccessKeyMetadata']:
+        if access_key['AccessKeyId'] == self.id:
+            response = access_key
```

Comment:
Is it a safe assumption that we'll only ever have one access key in the response? Otherwise, this always returns the last one. It will also break iteration because it mutates response while we're in the loop. We should be storing the final result in something else.

Reply:
There could be more than one access key in the response. That's why i am using if statement to get only the response of the key id we want. But every response inside ```AccessKeyMetadata``` would have different key id. 

Yes, agree changed the final result value to store in a different variable.